zebrad/src/commands.rs

```diff
@@ -68,15 +68,17 @@ impl Configurable<ZebradConfig> for ZebradCmd {
     fn config_path(&self) -> Option<PathBuf> {
         let if_exists = |f: PathBuf| if f.exists() { Some(f) } else { None };
 
-        dirs::preference_dir()
-            .map(|path| path.join(CONFIG_FILE))
-            .and_then(if_exists)
-            .or_else(|| std::env::current_dir().ok())
-            .map(|path| path.join(CONFIG_FILE))
-            .and_then(if_exists)
-
         // Note: Changes in how configuration is loaded may need usage
         // edits in generate.rs
+        std::env::current_dir()
+            .ok()
+            .map(|path| path.join(CONFIG_FILE))
+            .and_then(if_exists)
+            .or_else(|| {
+                dirs::preference_dir()
+                    .map(|path| path.join(CONFIG_FILE))
+                    .and_then(if_exists)
```

Comment:
What is the reason for this change?

We have changed the order of current and preference dirs a few times. So let's think carefully about the reasons for the design, and document our design decision in a comment.

If there is no command-line argument, then zebrad has to guess which config the user wants. It seems like most users would want:
1. The config they have specifically placed in the preference directory, which *does not* depend on the directory they launch zebrad from.
2. The config that happens to be in the current directory they launched zebrad from.
3. The default config.

Reply:
I don't think we should change this zebrad behaviour just for the tests.

Instead, we can pass `-c zebrad.toml` to each test.

Reply:
@teor2345 so the scenario that made me change it is when you're starting multiple tests, and you're configuring them by constructing a configuration file in their working directory and spawning them in different working directories, but you've got a configuration setup in the main default directory that they'd pick via `preference_dir`. The old logic would prioritize `explicit config path > implied default location > implied current directory`. I think when both exist we should probably prefer the most local configuration file, which is what this change does.

Reply:
I'd expect zebrad to pick the config that is most specific, which to me is `config in current dir > config in default dir > default config`. If both options 1 and 2 are valid then I think 2 is the more intuitive first choice.

Reply:
It seems like any outcome here would surprise some users.

Should we simplify the design?

We can ignore the config in the current directory, unless the user specifies it using `-c zebrad.toml`.

Reply:
Here's a similar conversation from the last PR:
https://github.com/ZcashFoundation/zebra/pull/714#discussion_r458529306
https://github.com/ZcashFoundation/zebra/pull/714#discussion_r458529306

Using the current directory for the config is non-standard. And it seems to be more confusing than I expected.

I think it will be simpler for users to put a config in the one default location, or `-c config.toml`.

Reply:
I think that's a good compromise, at least that way we don't end up with tests that silently depend on this subtle behaviour of which config is where, all the tests will need to be explicit about their config.